### PR TITLE
homework 5

### PR DESCRIPTION
**Pull request by AleksandrAlex**
Hello Aleksander, 
Please, check out my homework : )

### CODE REVIEW

File: app/src/main/java/ru/suslovalex/androidacademyapp/FragmentMoviesList.kt [hunk 1]
```diff
@@ -26,17 +30,29 @@ class FragmentMoviesList : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        val recycler: RecyclerView = view.findViewById(R.id.rv_moviesList)
-        adapterMovies = AdapterMovies({ id -> doOnClick(id) }, getListMovies())
-        recycler.layoutManager = GridLayoutManager(view.context, 2)
-        recycler.adapter = adapterMovies
-        recycler.hasFixedSize()
+        lifecycleScope.launch(Dispatchers.IO) {
```
Comment (alexander-smityuk) [inline app/src/main/java/ru/suslovalex/androidacademyapp/FragmentMoviesList.kt]: @AleksandrAlex No need `Dispatchers.IO` here, you have it inside` LoadJson`

File: app/src/main/java/ru/suslovalex/androidacademyapp/FragmentMoviesList.kt [hunk 2]
```diff
@@ -26,17 +30,29 @@ class FragmentMoviesList : Fragment() {
...
+            updateUI(view, movieList)
+        }
+    }
+
+    private suspend fun getMovies() = withContext(Dispatchers.IO){
+        movieList = loadMovies(this@FragmentMoviesList.requireContext())
     }
 
-    private fun doOnClick(id: Int) {
+    private suspend fun updateUI(view: View, movieList: List<Movie>) =
```
Comment (alexander-smityuk) [inline app/src/main/java/ru/suslovalex/androidacademyapp/FragmentMoviesList.kt]: @AleksandrAlex Why this fun is `suspend`, you can do this operation without coroutines

File: app/src/main/java/ru/suslovalex/androidacademyapp/FragmentMoviesDetails.kt [hunk 3]
```diff
@@ -64,30 +66,60 @@ class FragmentMoviesDetails : Fragment() {
...
-            currentMovie = getListMovies()[id]
-        }
+//    private fun hasActors(view: View) {
+//        if (currentMovie.actors[0]==null ){
+//
+//        }
+//    }
+
+    private fun getMovie() {
+        currentMovie = arguments?.getParcelable("movie")!!
```
Comment (alexander-smityuk) [inline app/src/main/java/ru/suslovalex/androidacademyapp/FragmentMoviesDetails.kt]: @AleksandrAlex Please avoid `!!`

File: app/src/main/java/ru/suslovalex/androidacademyapp/data/JsonLoad.kt [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    val overview: String,
+    val adult: Boolean
+)
+
+private suspend fun loadGenres(context: Context): List<Genre> = withContext(Dispatchers.IO) {
+    val data = readAssetFileToString(context, "genres.json")
+    parseGenres(data)
+}
+
+internal fun parseGenres(data: String): List<Genre> {
```
Comment (alexander-smityuk) [inline app/src/main/java/ru/suslovalex/androidacademyapp/data/JsonLoad.kt:50]: @AleksandrAlex Please use coroutines for parsing too. You need to do it on `Dispachers.Default`

**Review by alexander-smityuk — COMMENTED** [review]
Looks good. Please fix the comments and push again.

File: app/src/main/java/ru/suslovalex/androidacademyapp/data/JsonLoad.kt (hunk 4 above)
Comment (AleksandrAlex) [reply app/src/main/java/ru/suslovalex/androidacademyapp/data/JsonLoad.kt:50]: JsonLoad is not my file! Should I edit it???

File: app/src/main/java/ru/suslovalex/androidacademyapp/FragmentMoviesDetails.kt [hunk 5]
```diff
@@ -69,50 +66,37 @@ class FragmentMoviesDetails : Fragment() {
...
         val cast:TextView = view.findViewById(R.id.txt_cast)
 
-       
-
-
-        Glide.with(view).load(currentMovie.backdrop).into(image)
+        Glide.with(view).load(currentMovie?.backdrop).into(image)
         adult.text = getAgeForLabel(currentMovie)
-        title.text = currentMovie.title
+        title.text = currentMovie?.title
```
Comment (alexander-smityuk) [inline app/src/main/java/ru/suslovalex/androidacademyapp/FragmentMoviesDetails.kt]: @AleksandrAlex You can use `currentMovie?.let` once and not check it again

**Review by alexander-smityuk — APPROVED** [review]

**Pull request merged**